### PR TITLE
add PR comments

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**What this PR does / why we need it**:
Add comments to PRs explaining what happens.
In the second part, the Build URL will be also added as comment

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: whd/dispatcher.py [hunk 1]
```diff
@@ -226,13 +241,28 @@ def _set_pr_labels(self, pr_event, resources) -> bool:
                     f"Update to pull request #{pr_event.number()} by org member '{sender_login}' "
                     f" in '{repository_path}' found. Setting required labels '{required_labels}'."
                 )
+                github_helper.add_comment_to_pr(
+                    pull_request_number=pr_number,
+                    comment=(
+                        "Thank you for updating the PR. I will start a build. Once started the "
```
Comment (ccwienk) [inline whd/dispatcher.py:247]: nit: comma after `Once started`?

File: whd/dispatcher.py [hunk 2]
```diff
@@ -205,34 +205,66 @@ def _set_pr_labels(self, pr_event, resources) -> bool:
...
         if pr_event.action() is PullRequestAction.OPENED:
             if github_helper.is_pr_created_by_org_member(pr_number):
                 logger.info(
                     f"New pull request by member of '{owner}' in '{repository_path}' found. "
                     f"Setting required labels '{required_labels}'."
                 )
+                github_helper.add_comment_to_pr(
+                    pull_request_number=pr_number,
+                    comment=(
+                        f"Thank you {sender_login} for your contribution. I will start a build for "
```
Comment (AndreasBurger) [inline whd/dispatcher.py:218]: Not sure whether this'd work, but consider using @{sender_login}. (Also for all the other occurrences)

File: concourse/steps/rm_pr_label.mako [hunk 3]
```diff
@@ -25,6 +27,18 @@ with open(pr_id_path) as f:
...
+build_url = own_running_build_url()
+github_helper = GitHubRepositoryHelper(
+    owner='${main_repo.repo_owner()}',
+    name='${main_repo.repo_name()}',
+    github_cfg=github_cfg,
+)
+github_helper.add_comment_to_pr(
+    pull_request_number=pr_id,
+    comment=(
+        f"PR build URL: {build_url}"
```
Comment (ccwienk) [inline concourse/steps/rm_pr_label.mako:39]: maybe make this a bit more "friendly", e.g. "Your change is being tested - visit {url} for your build and test log"?

File: concourse/util.py [hunk 4]
```diff
@@ -23,6 +23,7 @@
 import concourse.steps.meta
 import github.webhook
 
+from urllib.parse import urljoin
```
Comment (ccwienk) [inline concourse/util.py:26]: please use ci.util.urljoin - it is the better urljoin..

File: concourse/steps/rm_pr_label.mako (hunk 3 above)
Comment (AndreasBurger) [inline concourse/steps/rm_pr_label.mako:39]: Is is rather short, especially compared to the other comments. Consider something like

> A build of this pull request has started. You can check on its progress here: <url>

or similar

File: concourse/util.py [hunk 5]
```diff
@@ -158,6 +159,27 @@ def get_pipeline_metadata():
...
+        'teams',
+        pipeline_metadata.team_name,
+        'pipelines',
+        pipeline_metadata.pipeline_name,
+        'jobs',
+        pipeline_metadata.job_name,
+        'builds',
+        own_build.build_number()
+    ]
+    return urljoin(concourse_cfg.external_url(), '/'.join(url_parts))
```
Comment (ccwienk) [inline concourse/util.py:180]: use ci.util.urljoin - it is the better urljoin^^

File: concourse/util.py [hunk 6]
```diff
@@ -158,6 +159,27 @@ def get_pipeline_metadata():
     )
 
 
+def own_running_build_url():
+    if not _running_on_ci():
+        raise RuntimeError('Can only find own running build URL if running on CI infrastructure.')
+
+    pipeline_metadata = get_pipeline_metadata()
+    config_set = ctx().cfg_factory().cfg_set(pipeline_metadata.current_config_set_name)
```
Comment (ccwienk) [inline concourse/util.py:167]: pass-in cfg, please. no magic grabbing for globals, please..

File: concourse/util.py [hunk 7]
```diff
@@ -158,6 +159,27 @@ def get_pipeline_metadata():
     )
 
 
+def own_running_build_url():
+    if not _running_on_ci():
```
Comment (ccwienk) [inline concourse/util.py:163]: if passing-in concourse-cfg, this check is not necessary

File: concourse/steps/rm_pr_label.mako [hunk 8]
```diff
@@ -1,5 +1,5 @@
 <%def
-  name="rm_pr_label_step(job_step, job_variant, github_cfg, indent)",
+  name="rm_pr_label_step(job_step, job_variant, cfg_set, indent)",
```
Comment (ccwienk) [inline concourse/steps/rm_pr_label.mako:2]: please rather pass github_cfg && concourse_cfg explicitly.

reasoning: this way, we do not implicitly depend on the default-cfg element from cfg-set, but accept a more explicit cfg, which makes the function more generic/flexible on the one-hand-side, and easier to understand (imo..) -> explicit is better than implicit

**Pull request merged**